src/VASSAL/build/module/map/PieceMover.java

```diff
@@ -441,25 +441,39 @@ protected boolean isMultipleSelectionEvent(MouseEvent e) {
 
   /** Invoked just before a piece is moved */
   protected Command movedPiece(GamePiece p, Point loc) {
-    setOldLocation(p);
-    Command c = null;
+    Command c = new NullCommand();
+    c = c.append(setOldLocations(p));
     if (!loc.equals(p.getPosition())) {
-      c = markMoved(p, true);
+      c = c.append(markMoved(p, true));
     }
     if (p.getParent() != null) {
       final Command removedCommand = p.getParent().pieceRemoved(p);
-      c = c == null ? removedCommand : c.append(removedCommand);
+      c = c.append(removedCommand);
     }
     return c;
   }
-
+  
+  /**
+   * Replaced by {@link #setOldLocations(GamePiece)} to return generated Commands
+   * @param p
+   */
+  @Deprecated
   protected void setOldLocation(GamePiece p) {
+    setOldLocations(p);
```

Comment:
So, what will happen now if the old function is called? The new function returns a `Command`, which appears to be thrown away?

Reply:
Yes, that's right. The Old function will only be called by existing custom code subclassing the PieceMover. They are not expecting a Command to be returned.  The Command to be returned is built up in a local variable and there is no way we can 'make' them use our new Commands. The best we can do is keep the old code working in the old broken way.